pylint/checkers/classes.py

```diff
@@ -1841,6 +1850,14 @@ def _check_len(self, node):
         if not isinstance(value, int) or value < 0:
             self.add_message("invalid-length-returned", node=node)
 
+    def _check_bool(self, node):
+        inferred = _safe_infer_call_result(node, node)
+        if not inferred or inferred is astroid.Uninferable:
+            return
+
+        if not isinstance(inferred, astroid.Instance) or inferred.name != "bool":
```

Comment:
You can use a `Const` node to check if it returns bools or not:
```suggestion
        if not isinstance(inferred, astroid.Const) or not isinstance(inferred.value, bool):
                ....
```

Reply:
Thanks! I'll update my changes!